### ISSUE TITLE
Window snaps to monitor when resizing

### ISSUE THREAD

**Issue by soufhehe** [hyprwm/Hyprland#8554]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland 0.45.0 built from branch main at commit 54f57797e9d025db72777ccf4adb0ddb25125016  (snap: account for position of multiple monitors (8543)).
Date: Sat Nov 23 15:36:28 2024
Tag: v0.45.0-59-g54f57797, commits: 5492
built against aquamarine 0.5.0


flags set:
debug


System Information:
System name: Linux
Node name: aphe-pc
Release: 6.12.1-2-cachyos
Version: #1 SMP PREEMPT_DYNAMIC Fri, 22 Nov 2024 17:05:17 +0000


GPU information: 
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Renoir [Radeon Vega Series / Radeon Vega Mobile Series] [1002:1636] (rev c2) (prog-if 00 [VGA controller])


os-release: NAME="CachyOS Linux"
PRETTY_NAME="CachyOS"
ID=cachyos
ID_LIKE=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://cachyos.org/"
DOCUMENTATION_URL="https://wiki.cachyos.org/"
SUPPORT_URL="https://discuss.cachyos.org/"
BUG_REPORT_URL="https://github.com/cachyos"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=cachyos


plugins:
	unknown: not runtime


```

</details>

### Description

If general:monitor_gap is too big, windows snap to monitor when resizing

### How to reproduce

monitor = eDP-1, addreserved, 0, 0, 100, 0
    snap {
    	 enabled = true
	 monitor_gap = 101
    }

resize the window when it's stuck in the leftmost window

### Crash reports, logs, images, videos

https://github.com/user-attachments/assets/3966cb57-9cf0-40ae-ae17-c793dd8be726
sorry for the sound

**Comment by myQwil** [comment]
I should be able to fix this, I just need to add checks for which corner is being grabbed, as is done between floating windows.

**Comment by myQwil** [comment]
Fix is now here: https://github.com/hyprwm/Hyprland/pull/8637

**Comment by soufhehe** [comment]
Thanks !

**Comment by soufhehe** [comment]
https://github.com/hyprwm/Hyprland/pull/8637#issuecomment-2533082260

**Comment by myQwil** [comment]
@soufhehe A fix has been proposed here: https://github.com/hyprwm/Hyprland/pull/8694
Let me know if that solution works for you. Here's how it looks on my end:

https://github.com/user-attachments/assets/0a976d4c-031f-4cf3-853d-acbac2678a7b

**Comment by soufhehe** [comment]
> @soufhehe A fix has been proposed here: #8694 Let me know if that solution works for you. Here's how it looks on my end:
> 2024-12-11.00-29-54.mp4

It works exactly as I'd expect it ! Thank you again !
